Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -1131,29 +1141,31 @@ class CombinatorialKalmanFilter {
   /// the filter and smoother
   ///
   /// @tparam source_link_container_t Source link container type
-  /// @tparam start_parameters_t Type of the initial parameters
+  /// @tparam start_parameters_container_t Initial parameters container type
   /// @tparam calibrator_t Type of the source link calibrator
+  /// @tparam measurement_selector_t Type of the measurement selector
   /// @tparam parameters_t Type of parameters used for local parameters
   ///
   /// @param sourcelinks The fittable uncalibrated measurements
-  /// @param sParameters The initial track parameters
+  /// @param initialParameters The initial track parameters
   /// @param tfOptions CombinatorialKalmanFilterOptions steering the track
   /// finding
   /// @note The input measurements are given in the form of @c SourceLinks.
   /// It's
   /// @c calibrator_t's job to turn them into calibrated measurements used in
   /// the track finding.
   ///
-  /// @return the output as an output track
-  template <typename source_link_container_t, typename start_parameters_t,
-            typename calibrator_t, typename source_link_selector_t,
+  /// @return a container of output tracks
+  template <typename source_link_container_t,
+            typename start_parameters_container_t, typename calibrator_t,
+            typename measurement_selector_t,
             typename parameters_t = BoundTrackParameters>
-  Result<CombinatorialKalmanFilterResult<
+  std::vector<CombinatorialKalmanFilterResult<
       typename source_link_container_t::value_type>>
   findTracks(const source_link_container_t& sourcelinks,
```

Comment:
That's a good question. The results for all seeds are returned now (though this might not be a good solution).

Reply:
It gives the possibility to check the return value for all input seeds but it also allocates quite a bit of memory that is only used temporarily. It might be Ok for now, but I suggest that this be discussed in tomorrows developers meeting.